granulate_utils/metadata/databricks_client.py

```diff
@@ -43,46 +50,80 @@ def get_webui_address() -> Optional[str]:
 
     def get_job_name(self) -> Optional[str]:
         # Retry in case of a connection error, as the metrics server might not be up yet.
-        for i in range(MAX_RETRIES):
-            time.sleep(10)
+        retries = 0
+        while retries < MAX_RETRIES:
+            retries += 1
```

Comment:
Why are you implementing retry logic manually instead of using `backoff` ?

Reply:
Not required IMO for this PR.